fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexMergePolicy.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * IndexMergePolicy.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+/**
+ * Some store's indexes may need merging on some occasions. This helper module should allow the caller
+ * to set and prob the merge policy and merge requests.
+ */
+public class IndexMergePolicy {
+    private boolean indexNeedsMerge = false;
+    private boolean mergeDuringClose = true;
+
+    public boolean isIndexNeedsMerge() {
```

Comment:
With the current implementation, setting the values of the booleans is not thread safe, in the sense that under different threads paths, the results can be inconsistent. It may not be a big deal now, as I guess that we're only ever setting indexNeedsMerge to TRUE and mergeDuringClose to FALSE, but in the future it can be an opening to bugs. 
One way to fix this is to, for example, only allow "flipping" the value and not actually setting it to any boolean you want.

Reply:
Sure.